### ISSUE TITLE
Slack settings form edit bug

### ISSUE THREAD

**Issue by marcusos** [metabase/metabase#4673]
Your browser and the version: Firefox 52
    Your operating system: Windows 10
    Your databases: Google SQL (MySQL 5.7)
    Metabase version: 0.23.0
    Metabase hosting environment: Google App Engine
    Metabase internal database: Google SQL  (MySQL 5.7)

I tried to add slack integration, but when I clicked 'save changes' I got the following error :

Duplicate entry 'slack-token' for key 'PRIMARY'

![bug_app_engine](https://cloud.githubusercontent.com/assets/4155254/24623211/07768852-187e-11e7-9d06-b54cfcb01eae.PNG)

Logs :
```json
{
    "message": "Duplicate entry 'slack-token' for key 'PRIMARY'",
    "stacktrace": [
        "models.setting$fn__15362$set_string_BANG___15367$fn__15368.invoke(setting.clj:188)",
        "models.setting$fn__15362$set_string_BANG___15367.invoke(setting.clj:173)",
        "models.setting$set_BANG_.invokeStatic(setting.clj:227)",
        "models.setting$set_BANG_.invoke(setting.clj:218)",
        "models.setting$set_many_BANG_.invokeStatic(setting.clj:330)",
        "models.setting$set_many_BANG_.invoke(setting.clj:323)",
        "api.slack$fn__34095$fn__34098$fn__34099.invoke(slack.clj:23)",
        "api.slack$fn__34095$fn__34098.invoke(slack.clj:19)",
        "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:229)",
        "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:224)",
        "api.slack$fn__34095.invokeStatic(slack.clj:11)",
        "api.slack$fn__34095.invoke(slack.clj:11)",
        "middleware$enforce_authentication$fn__34556.invoke(middleware.clj:119)",
        "api.routes$fn__34720.invokeStatic(routes.clj:78)",
        "api.routes$fn__34720.invoke(routes.clj:78)",
        "routes$fn__35755.invokeStatic(routes.clj:44)",
        "routes$fn__35755.invoke(routes.clj:44)",
        "middleware$log_api_call$fn__34655$fn__34657.invoke(middleware.clj:331)",
        "middleware$log_api_call$fn__34655.invoke(middleware.clj:330)",
        "middleware$add_security_headers$fn__34605.invoke(middleware.clj:246)",
        "middleware$bind_current_user$fn__34560.invoke(middleware.clj:139)",
        "middleware$maybe_set_site_url$fn__34609.invoke(middleware.clj:268)"
    ],
    "sql-exception-chain": [
        "MySQLIntegrityConstraintViolationException:",
        "Message: Duplicate entry 'slack-token' for key 'PRIMARY'",
        "SQLState: 23000",
        "Error Code: 1062"
    ]
}
```

**Comment by camsaul** [comment]
@marcusos are you running more than one Metabase server?

**Comment by marcusos** [comment]
@camsaul No, It's just one Metabase server.

**Comment by camsaul** [comment]
@marcusos not sure why it's happening but if you restart your Metabase server it should fix the issue

**Comment by marcusos** [comment]
I am having a similar bug trying to set up a sendgrid email configuration, like this issue #4178

**Comment by camsaul** [comment]
Yeah. This looks like the same underlying issue as #4178.

I'm going to close this as a duplicate of that. I'll take a look at it I think I know what's going on

**Comment by camsaul** [comment]
Fixed by #4674, going out in our next release
